src/Nethermind/Nethermind.Evm/EvmStack.cs

```diff
@@ -168,39 +167,9 @@ public void PushUInt256(ref UInt256 value)
             }
         }
 
-        public void PushSignedInt(in BigInteger value)
+        public void PushSignedInt256(in Int256.Int256 value)
         {
-            int sign = value.Sign;
-            if (sign == 0)
-            {
-                PushZero();
-                return;
-            }
-
-            Span<byte> word = _bytes.Slice(Head * 32, 32);
-            if (sign > 0)
-            {
-                word.Clear();
-            }
-            else
-            {
-                word.Fill(0xff);    
-            }
-
-            Span<byte> fullBytes = stackalloc byte[33];
-            value.TryWriteBytes(fullBytes, out int bytesWritten, false, true);
-            int fillCount = 32 - bytesWritten;
-            int start = bytesWritten == 33 ? 1 : 0;
-            int length = bytesWritten == 33 ? 32 : bytesWritten;
-            fullBytes.Slice(start, length).CopyTo(fillCount > 0 ? word.Slice(fillCount, 32 - fillCount) : word);
-
-            if (_tracer.IsTracingInstructions) _tracer.ReportStackPush(word);
-
-            if (++Head >= MaxStackSize)
-            {
-                Metrics.EvmExceptions++;
-                throw new EvmStackOverflowException();
-            }
+            PushUInt256((UInt256) value);
```

Comment:
Not sure whats going on here

Reply:
replacing PushBigInteger with the new code - diff tool confused you